src/middlewared/middlewared/plugins/jail.py

```diff
@@ -111,59 +114,77 @@ def query(self, filters=None, options=None):
         return await self.middleware.call('jail.create_job', options)
 
     @private
-    @accepts(
-        Dict("options",
-             Str("release", required=True),
-             Str("template"),
-             Str("pkglist"),
-             Str("uuid"),
-             Bool("basejail", default=False),
-             Bool("empty", default=False),
-             Bool("short", default=False),
-             List("props", default=[])))
     @job()
     def create_job(self, job, options):
-
         verrors = ValidationErrors()
 
-        self.validate_ips(verrors, options)
+        try:
+            iocage = ioc.IOCage(skip_jails=True, jail=options['uuid'])
+            iocage.__check_jail_existence__()
+        except SystemExit:
```

Comment:
I am not sure SystemExit is the right exception here. The exception differs when running iocage in foreground or as a daemon.

Is that right @skarekrow

Reply:
Yeah good catch. It should be a CallError or something. iocage does different exceptions internally and this is external.